src/System.Management.Automation/namespaces/FileSystemProvider.cs

```diff
@@ -7845,6 +7849,39 @@ private struct REPARSE_GUID_DATA_BUFFER
             FileAttributes dwFlagsAndAttributes,
             IntPtr hTemplateFile);
 
+        [DllImport(PinvokeDllNames.FindFirstFileDllName, EntryPoint = "FindFirstFileExW", SetLastError = true, CharSet = CharSet.Unicode)]
+        private static extern SafeFileHandle FindFirstFileEx(string lpFileName, FINDEX_INFO_LEVELS fInfoLevelId, ref WIN32_FIND_DATA lpFindFileData, FINDEX_SEARCH_OPS fSearchOp, IntPtr lpSearchFilter, int dwAdditionalFlags);
```

Comment:
This returns a SafeFileHandle, which eventually calls CloseHandle.
According to the FindFirstFileExW documentation, the handle should be closed with FindClose, not CloseHandle.

Reply:
Nice catch!  Will submit a new PR to address.

Reply:
https://github.com/PowerShell/PowerShell/pull/10405